### ISSUE TITLE
Form submits twice even with e.preventDefault()

### ISSUE THREAD

**Issue by SigalZ** [formvalidation/support#159]
Hi,
When I submit a form the first time, it submits once. 
Filling the same form again, submits it twice, then 3, 4 etc. times. Each time I fill the form again it adds another submission to it.

My code:
`$('.addToCartBtn').on('click', function() {
        product_id = $(this).data('id');
        weight = $(this).data('weight');

        $('#form_'+weight+'_'+product_id).formValidation({
            framework: 'bootstrap',
            icon: {
                valid: 'glyphicon glyphicon-ok',
                invalid: 'glyphicon glyphicon-remove',
                validating: 'glyphicon glyphicon-refresh'
            },
            fields: {
                qty: {
                    validators: {
                        notEmpty: { message: "Please enter the quantity" }
                    }
                },
                ground: {
                    validators: {
                        notEmpty: { message: "Please select the ground" }
                    }
                }
            }
        }).on('success.form.fv', function(e) {
            // Prevent form submission
            e.preventDefault();
        });

        $('#form_'+weight+'_'+product_id).submit(function(e) {
            var $form = $(e.target);

            // Use Ajax to submit form data
            $.ajax({
                url: '/modules/shopping_cart/ajax/add_to_cart.ajax.php',
                type: 'POST',
                data: $form.serialize(),
                success: function(result) {
                    // ... Process the result ...
                    $('#cartItems').html('<b>' + result + '</b>');

                    bootbox.dialog({
                        message: "The coffee has been added to your cart.",
                        title: "<span class='dialogSuccess'>Success</span>",
                        buttons: {
                            continueshopping: {
                                label: "Continue Shopping",
                                className: "btn-default",
                                callback: function() {
                                    $('#form_'+weight+'_'+product_id)[0].reset();

                                    // Reset the form
                                    $form.formValidation('resetForm');
                                }
                            },
                            showcart: {
                                label: "Go to Cart",
                                className: "btn-success",
                                callback: function() { }
                            }
                        }
                    });
                }
            });
        });
    });`

Can you please help?

Thank you

**Comment by nghuuphuoc** [comment]
I recommend to use [this approach](http://formvalidation.io/examples/ajax-submit/#using-ajax-to-submit-form-data) when using Ajax to submit the form data.

**Comment by SigalZ** [comment]
Thanks, I changed my code but it still submits multiple times.

Am I missing something?

My code now:

` $('#form_'+weight+'_'+product_id).formValidation({
            framework: 'bootstrap',
            icon: {
                valid: 'glyphicon glyphicon-ok',
                invalid: 'glyphicon glyphicon-remove',
                validating: 'glyphicon glyphicon-refresh'
            },
            fields: {
                qty: {
                    validators: {
                        notEmpty: { message: "Please enter the quantity" }
                    }
                },
                ground: {
                    validators: {
                        notEmpty: { message: "Please select the ground" }
                    }
                }
            }
        }).on('success.form.fv', function(e) {
            // Prevent form submission
            e.preventDefault();

            var $form = $(e.target);

            // Use Ajax to submit form data
            $.ajax({
                url: '/modules/shopping_cart/ajax/add_to_cart.ajax.php',
                type: 'POST',
                data: $form.serialize(),
                success: function(result) {
                    // ... Process the result ...
                    $('#cartItems').html('<b>' + result + '</b>');

                    bootbox.dialog({
                        message: "The coffee has been added to your cart.",
                        title: "<span class='dialogSuccess'>Success</span>",
                        buttons: {
                            continueshopping: {
                                label: "Continue Shopping",
                                className: "btn-default",
                                callback: function() {
                                    $('#form_'+weight+'_'+product_id)[0].reset();

                                    // Reset the form
                                    $form.formValidation('resetForm');
                                }
                            },
                            showcart: {
                                label: "Go to Cart",
                                className: "btn-success",
                                callback: function() { }
                            }
                        }
                    });
                }
            });
        });
    });`

**Comment by nghuuphuoc** [comment]
Is `$('#form_'+weight+'_'+product_id).formValidation()` called inside `addToCartBtn` button click handler?

**Comment by nghuuphuoc** [comment]
Accoriding to the code from your first comment, because the `.on('success.form.fv')` is handler inside the `addToCartBtn` button click handler, the event will be triggered multiple times from the second, third, ... time you click the button.

To solve this, you can destroy the formValidation instance (so, the previous `success.form.fv` handler will be removed) before creating new one:

```javascript
$('.addToCartBtn').on('click', function() {
    product_id = $(this).data('id');
    weight = $(this).data('weight');

    $('#form_'+weight+'_'+product_id)
        .formValidation('destroy') // Add this
        .formValidation({
            ... options ...
        })
        .on('success.form.fv', function(e) {
            ...
        });
});
```

PS: I haven't tried this, but you also can removed the previous `success.form.fv` handler  by calling `.off('success.form.fv')`:

```javascript
$('.addToCartBtn').on('click', function() {
    product_id = $(this).data('id');
    weight = $(this).data('weight');

    $('#form_'+weight+'_'+product_id)
        .formValidation({
            ... options ...
        })
        .off('success.form.fv') // Hey, removed the previous handler
        .on('success.form.fv', function(e) {
            ...
        });
});
```

**Comment by SigalZ** [comment]
Thank you very much, that solved the problem.

**Comment by nghuuphuoc** [comment]
Thank for your confirmation. I updated the [Form is submitted twice](http://formvalidation.io/examples/form-submit-twice/#ensure-that-successformfv-is-triggered-once) example.
Hopefully it will help other users.
